cranelift/frontend/src/frontend.rs

```diff
@@ -768,9 +768,9 @@ impl<'a> FunctionBuilder<'a> {
 
             let ch = u64::from(ch);
             let raw_value = if int_type == types::I64 {
-                (ch << 32) | (ch << 16) | (ch << 8) | ch
+                ch * 0x0101010101010101_u64
             } else if int_type == types::I32 {
-                (ch << 16) | (ch << 8) | ch
+                ch * 0x01010101_u64
```

Comment:
Could you keep using shift and bitor?

Reply:
Yes, sure!

Reply:
But keep in mind LLVM can't optimize such series of shifts:
https://godbolt.org/z/nadxYq

Reply:
I assumed that shift and bitor would be faster than multiplying, but according to llvm-mca, multiplying is faster for 32bit and 64bit ints.

Reply:
> I assumed that shift and bitor would be faster than multiplying, but according to llvm-mca, multiplying is faster for 32bit and 64bit ints.

No. 64-bit multiply will be always faster in this case. Even for 32-bit platforms.

btw trunk branck of LLVM already perform this optimization but still suboptimal. See:
https://godbolt.org/z/o3jfWv

Reply:
Ok. Could you please revert back to multiplying?